smartsim/_core/mli/infrastructure/control/error_handling.py

```diff
@@ -63,10 +66,13 @@ def exception_handler(
     :param reply_channel: The channel used to send replies
     :param failure_message: Failure message to log and send back
     """
+    if failure_message is None:
+        failure_message = str(exc)
     logger.exception(
```

Comment:
What is missing from a standard `logger.exception` output that causes us to have this custom format string? doesn't this result in duplication in the final message? Here's a quick example showing the traceback logged twice, the exception type included twice, etc:

```
>>> try:
...  raise_ex()
... except Exception as ex:
...  logger.exception(f"Exception type: {type(ex).__name__}\nTraceback: {traceback.format_exc()}")
... 
ERROR:root:Exception type: Exception
Traceback: Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<stdin>", line 2, in raise_ex
Exception
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<stdin>", line 2, in raise_ex
Exception
```

Reply:
No you're totally right. This started out as me wanted to ensure that our failure message gets logged with the exception but now it's just kind of crazy. I think I just need to simply do 
```
logger.exception(exc)
```

but still set the `failure_message` to `str(exc)` if there wasn't one passed in so we can send it back in the response.